analogway_livecore.js

```diff
@@ -1,743 +1,564 @@
+							break
+						case 117:
+							this.modelname = 'NeXtage 08 - 4K'
+							break
+						case 118:
+							this.modelname = 'Ascender 32 - 4K -PL'
+							break
+						case 119:
+							this.modelname = 'Output Expander 32 - 4K - PL'
+							break
+						default:
+							this.modelname = 'unknown'
+							break
+					}
+					this.log('info', this.config.label + ' Type is ' + this.modelname)
+					this.sendcmd('0,TPver')
+				}
 
-			if (line.match(/TPver\d+/)) {
-				var commandSetVersion = parseInt(line.match(/TPver\d+,(\d+)/)[1]);
-				self.log('info', "Command set version of " + self.config.label +" is " + commandSetVersion);
-				// TODO: Should check the machine state now, will be implemented after feedback system is done
-			}
+				if (line.match(/TPver\d+/)) {
+					var commandSetVersion = parseInt(line.match(/TPver\d+,(\d+)/)[1])
+					this.log('info', 'Command set version of ' + this.config.label + ' is ' + commandSetVersion)
+					// TODO: Should check the machine state now, will be implemented after feedback system is done
+				}
 
-			if (line.match(/TPdie0/)) {
-				//There is no parameter readback runnning, it can be started now
-			}
+				if (line.match(/TPdie0/)) {
+					//There is no parameter readback runnning, it can be started now
+				}
 
+				if (line.match(/E\d{2}/)) {
+					switch (parseInt(line.match(/E(\d{2})/)[1])) {
+						case 10:
+							this.log('error', 'Received command name error from ' + this.config.label + ': ' + line)
+							break
+						case 11:
+							this.log('error', 'Received index value out of range error from ' + this.config.label + ': ' + line)
+							break
+						case 12:
+							this.log(
+								'error',
+								'Received index count (too few or too much) error from ' + this.config.label + ': ' + line
+							)
+							break
+						case 13:
+							this.log('error', 'Received value out of range error from ' + this.config.label + ': ' + line)
+							break
+						default:
+							this.log('error', 'Received unspecified error from Livecore ' + this.config.label + ': ' + line)
+					}
+				}
 
-			if (line.match(/E\d{2}/)) {
-				switch (parseInt(line.match(/E(\d{2})/)[1])) {
-					case 10: self.log('error',"Received command name error from "+ self.config.label +": "+ line); break;
-					case 11: self.log('error',"Received index value out of range error from "+ self.config.label +": "+ line); break;
-					case 12: self.log('error',"Received index count (too few or too much) error from "+ self.config.label +": "+ line); break;
-					case 13: self.log('error',"Received value out of range error from "+ self.config.label +": "+ line); break;
-					default: self.log('error',"Received unspecified error from Livecore "+ self.config.label +": "+ line);
+				if (line.match(/TAopr\d,\d/)) {
+					//Program Tally Information
+					let prearr = line.split(',')
+					let input = [Number(prearr[0].replace('TAopr', '')), prearr[1]]
+					if (input[1] === '1') {
+						tallyPGM.push(input[0])
```

Comment:
Number may be pushed multiple times, e.g. if a tally of command gets not thru. Then the feedback is broken.
Better implementation would be `tallyPGM[input[0]] = input[1]`. Then you have the state of each input in the corresponding index. Reading out would give 0 or 1 or undefined if the tally haven't been set.